### ISSUE TITLE
[feature] Warn about `exit $?` after last command in bash script

### ISSUE THREAD

**Issue by NateEag** [koalaman/shellcheck#1187]
#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#! /bin/bash

ls -lh

exit $?

```

#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

Bash scripts exit with the status code of the last command, so this exit call is redundant.

#### Context

I learned today that Bash has this feature. As with so many bash features, I had no idea.

I was surprised that Shellcheck hadn't caught this redundancy for me.

**Comment by NateEag** [comment]
I've been curious about Haskell for a while, so if this looks like a check that should be added, I'd be interested in carving out some time to make a PR.

**Comment by NateEag** [comment]
That's a good point, @matthewpersico . Thanks for bringing it up.

**Comment by matthewpersico** [comment]
You’re welcome. If you still want to muck around with Haskell and you wanna do something that should be simple yet useful, how about trying to program in .shellcheckrc functionality? First thing I can think of that would be useful is allowing a list of disables that is always applied(instead of having to put them on the command line every time. ) see #725 and #323. :-)

**Comment by Dmole** [comment]
Actually the behaviors are different; if you call exit explicitly it will exit from a script that includes the file, otherwise not.

**Comment by NateEag** [comment]
Just noticed this in my personal issues list.

Since it is not actually a smart check to add, I'll mark this issue as closed.
